### PR TITLE
Restructure modules and class names

### PR DESCRIPTION
**Pull request by llucax**
This PR do some minor cleanups and some major project structuring to use more consistent and shorter modules and class names.

* `frequenz.client.base.grpc_streaming_helper` was renamed to `frequenz.client.base.streaming`.

   - The `GrpcStreamingHelper` class was renamed to `GrpcStreamBroadcaster`.

      + The constructor argument `retry_spec` was renamed to `retry_strategy`.

* `frequenz.client.base.retry_strategy` was renamed to `frequenz.client.base.retry`.

   - The `RetryStrategy` class was renamed to `Strategy`.

### CODE REVIEW

**Comment by llucax** [comment]
I'm planning to improve the docs too, but I will wait until this is merged because I can imagine there could be some push back. BTW, I really think we should eventually rename this project `frenquenz-grpc` or `frequenz-grpcutil` or something like that, this is not really a base class for clients but just utilities to deal with gRPC/protobuf.

**Review by Marenz — DISMISSED** [review]

**Review by Marenz — DISMISSED** [review]

File: src/frequenz/client/base/streaming.py
Comment (shsms) [inline src/frequenz/client/base/streaming.py]: `frequenz.client.base.streaming.Broadcaster` feels somewhat broad, for a thing that takes a streaming grpc call and puts the gprc messages into a channel.  If anything, it should be the name for an ABC, when there's such a need, but not to any one thing.

I like `GrpcStreamBroadcaster`, because for someone who sees it, it already gives some idea what it does.  But maybe there are other options.

File: src/frequenz/client/base/streaming.py
Comment (shsms) [inline src/frequenz/client/base/streaming.py:27]: Maybe the word Helper can go away from the docstring as well?

**Review by shsms — COMMENTED** [review]
Just a comment about the name.

File: src/frequenz/client/base/streaming.py
Comment (Marenz) [reply src/frequenz/client/base/streaming.py]: Grpcaster :eyes:

File: src/frequenz/client/base/streaming.py
Comment (llucax) [reply src/frequenz/client/base/streaming.py:27]: I was planning to update docs in a separate PR, I didn't want to have to rewrite all the docs if there was a lot of push back about the new names.

File: src/frequenz/client/base/streaming.py
Comment (llucax) [reply src/frequenz/client/base/streaming.py]: This is the point I expected to be controversial the most, because I wasn't also 100% convinced. I'm OK with `GrpcStreamBroadcaster` even if I find it a bit too verbose, at least in the context of my vision of having this repo renamed to `frequenz-grpcutil` and then using `from frequenz.grpcutil import streaming; b = streaming.GrpcStreamBroadcaster`.

But when in doubt, I definitely think more explicit names are better than more generic ones, so let's go for that. Will update.

**Comment by llucax** [comment]
I will update and improve the docs in a followup PR, and then do the v0.2.0 release.

**Comment by llucax** [comment]
Enabled auto-merge.

**Comment by shsms** [comment]
do you want to update the commit message also? https://github.com/frequenz-floss/frequenz-client-base-python/pull/25/commits/edab315116c3e0ed9ceb46f876ac6b664abd6163

**Comment by llucax** [comment]
Good catch, done now!

**Review by shsms — APPROVED** [review]

**Comment by shsms** [comment]
I updated the PR description also with the new name, and removed and re-enabled auto-merge, so that it uses the updated merge commit message.

**Pull request merged**
